jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-kops-aws.yaml

```diff
@@ -0,0 +1,67 @@
+- job-template:
+    name: 'kubernetes-e2e-kops-{suffix}'
+    node: '{jenkins_node}'
+    description: '{description} Test owner: {test-owner}.'
+    jenkins_node: 'e2e'
+    disabled: '{obj:disable_job}'
+    properties:
+        - build-discarder:
+            days-to-keep: 7
+    triggers:
+        - reverse:
+            jobs: '{trigger-job}'
+            result: success
+        - timed: '{cron-string}'
+    publishers:
+        - junit-publisher
+        - email-ext:
+            recipients: '{emails}'
+        - gcs-uploader
+    provider-env: |
+        export E2E_OPT="--deployment kops --kops $WORKSPACE/kops -kops-cluster {suffix}.test-aws.k8s.io -kops-state s3://k8s-kops-jenkins/"
+        export AWS_CONFIG_FILE=/workspace/.aws/credentials
+        # This is needed to be able to create PD from the e2e test
+        export AWS_SHARED_CREDENTIALS_FILE=/workspace/.aws/credentials
+        export KUBE_SSH_USER=admin
+    builders:
+        - activate-gce-service-account
+        - shell: |
+            {provider-env}
+            {job-env}
+            {post-env}
+            if ! curl -fsS --retry 3 -o "$WORKSPACE/kops.url" "https://storage.googleapis.com/kops-ci/bin/latest-ci.txt"; then
```

Comment:
As for what I mean by error checking, this section of bash is easily rewritten as:
```
export KOPS_URL=$(curl -fsS --retry 3 https://storage.googleapis.com/kops-ci/bin/latest-ci.txt)"
curl -fsS --retry 3 -o "${KOPS_URL}/kops" "$(cat $WORKSPACE/kops.url)/linux/amd64/kops"
export NODEUP_URL="${KOPS_URL}/linux/amd64/nodeup"
```